zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/ContentTest.java

```diff
@@ -0,0 +1,275 @@
+
+        when(mockPermissionsService.canEdit(mockSession)).thenReturn(true);
+        when(mockPermissionsService.canView(any(), any())).thenReturn(true);
+
+        when(mockSessions.get()).thenReturn(mockSession);
+
+        when(mockCollectionKeyring.get(any(), any())).thenReturn(secretKey);
+
+        when(mockZebedee.getCollections()).thenReturn(collections);
+        when(mockZebedee.getSessions()).thenReturn(mockSessions);
+        when(mockZebedee.getEncryptionKeyFactory()).thenReturn(encryptionKeyFactory);
+        when(mockZebedee.getPermissionsService()).thenReturn(mockPermissionsService);
+        when(mockZebedee.getUsersService()).thenReturn(mockUsersService);
+        when(mockZebedee.getCollectionKeyring()).thenReturn(mockCollectionKeyring);
+        when(mockZebedee.getSlackNotifier()).thenReturn(mockNotifier);
+        Root.zebedee = mockZebedee;
+    }
+
+    @Test
+    public void WriteVersionFileV1() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        String collectionId = "aktesting";
+        request.setPathInfo("/content/" + collectionId);
+        String datasetURL = "/peoplepopulationandcommunity/birthsdeathsandmarriages/livebirths/datasets/babynamesenglandandwalesbabynamesstatisticsboys/2022/data.json";
+        request.addParameter("uri", datasetURL);
+        request.setContent(getV1RequestContent());
+
+        Path versionFile = createCollectionAndPaths(collectionId);
+
+        // Create the version content via the content API.
+        try {
+            Content contentApi = new Content();
+            boolean result = contentApi.saveContent(request, new MockHttpServletResponse());
+            Assert.assertTrue(result);
+        } catch (Exception e) {
+            Assert.fail(e.getMessage() + e.toString());
+            e.printStackTrace();
+            throw e;
+        }
+
+        byte[] bytes = IOUtils.toByteArray(EncryptionUtils.encryptionInputStream(versionFile, secretKey));
+        Assert.assertEquals(expectedContentV1, new String(bytes));
+
+        // read the version content via the content API.
+       MockHttpServletResponse response = new MockHttpServletResponse();
+        MockHttpServletRequest dataRequest = new MockHttpServletRequest();
+        dataRequest.setRequestURI("/data/" + collectionId);
+
+        String dataURL = "/peoplepopulationandcommunity/birthsdeathsandmarriages/livebirths/datasets/babynamesenglandandwalesbabynamesstatisticsboys/2022";
+        dataRequest.addParameter("uri", dataURL);
+
+        ZebedeeCollectionReaderFactory factory = new ZebedeeCollectionReaderFactory(mockZebedee);
+        ZebedeeReader.setCollectionReaderFactory(factory);
+
+        try {
+            Data dataAPI = new Data();
+            dataAPI.read(dataRequest, response);
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail(e.getMessage() + e.toString());
+            throw e;
+        }
```

Comment:
Actually, the IntelliJ test runner wasn't printing the exceptions' stack trace, so we added the explicit call. Also, the receipt of the exception is a test fail rather than a test error (which an exception from anywhere else would be).

Reply:
I'd argue that it is actually a test error since an exception is not expected to happen in the test?